networkit/cpp/graph/Graph.cpp

```diff
@@ -106,44 +107,67 @@ Graph::Graph(const Graph &G, bool weighted, bool directed, bool edgesIndexed)
         // G is directed, but we want an undirected graph
         // so we need to combine the out and in stuff for every node
         outEdges.resize(z);
+        if (weighted)
```

Comment:
If you change the order of loops, pls check performance changes for typical use cases.

Rearranging the loops leads to a different memory layout concerning the involved edge vectors. In the current scheme `edges`, `weights` and `ids` are contiguous each (with respect to iterating over nodes). After rearranging memory is sorted by nodes (`[edges_0, weights_0, ids_0, edges_1, weights_1, ...]`).

Reply:
The reason I changed the order of the loops is that when doing the weights and Ids we would have to check in the `out`/`inEdges` array to check if there is a collision between an inedge and an outedge anyway so that we wouldn't have continuous access of memory there even if we inserted each of them in their separate loop.